talk/objectorientation/adl.tex

```diff
@@ -168,3 +173,62 @@
     \end{cppcode*}
   \end{exampleblock}
 \end{frame}
+
+\begin{frame}[fragile]
+  \frametitlecpp[98]{Hidden Friends and ADL}
+  \begin{block}{Advantages of hidden friends}
+    \begin{itemize}
+      \item Are \emph{only} found via ADL, e.g.\ \mintinline{cpp}{std::cout << complex;}
+      \item Compiler needs to consider less functions during ordinary name lookup (faster compilation)
+      \item Avoid accidental implicit conversions
+    \end{itemize}
+  \end{block}
+  \begin{alertblock}{Accidental conversion example}
+    \small
+    \begin{cppcode*}{gobble=2}
+      std::ostream & operator<<(std::ostream & os,
+        Complex const & c) { ... } // out-of-class definition
+      // either: Complex::Complex(Fraction f) { ... }
```

Comment:
Why not put it as uncommented code inside the struct ?

Reply:
Because this is a constructor for `Complex`, which is not defined on this slide.